src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenUsingDeclarationTests.cs

```diff
@@ -1089,5 +1099,168 @@ static async Task Main()
             CompileAndVerify(source, expectedOutput: "Dispose; ");
         }
 
+        [Fact]
+        public void UsingDeclarationAsync()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+class C1 : IAsyncDisposable
+{
+    public ValueTask DisposeAsync() 
+    { 
+        Console.WriteLine(""Dispose async"");
+        return new ValueTask(Task.CompletedTask);
+    }
+}
+
+class C2
+{
+    static async Task Main()
+    {
+        await using C1 c = new C1();
+    }
+}";
+            var compilation = CreateCompilationWithTasksExtensions(source + _asyncDisposable, options: TestOptions.DebugExe).VerifyDiagnostics();
+
+            CompileAndVerify(compilation, expectedOutput: "Dispose async").VerifyIL("C2.Main", @"
```

Comment:
>VerifyIL("C2.Main" [](start = 75, length = 18)

I think this is not the method you intended to inspect.
`C2.Main` is an auto-generated wrapper for the real method. I expect to see a construction of `new C1()` and a `try/finally` in the IL.